File: 04-recursion/01-print-one-to-n/01-print-one-to-n.ts

const print_one_to_n = (n: number) => {
  const res = []
  const print = (n: number) => {
    if (n === 0) return
    print(n-1)
    res.push(n)
  }
  print(n)
  return res
}

console.log(print_one_to_n(5))

File: 04-recursion/01-print-one-to-n/01-print-one-to-n.py
def print_one_to_n(num):
  res = []
  def dfs_one_to_n(num):
    if num == 0:
      return
    dfs_one_to_n(num-1)
    res.append(num)
  dfs_one_to_n(num)
  return res

res = print_one_to_n(5)
print(res)